src/cli/main.rs

```diff
@@ -60,31 +81,17 @@ fn run() -> Result<ExitCode> {
                         }
                         None => {
                             let mode = ansi::get_colormode();
-
-                            if mode == ansi::Mode::Ansi8Bit
-                                && global_matches.subcommand_name() != Some("paint")
-                                && global_matches.subcommand_name() != Some("colorcheck")
-                            {
-                                write_stderr(Color::yellow(), "pastel warning",
-                                    "Your terminal emulator does not appear to support 24-bit colors \
-                                    (this means that the COLORTERM environment variable is not set to \
-                                    'truecolor' or '24bit'). \
-                                    pastel will fall back to 8-bit colors, but you will only be able \
-                                    to see rough approximations of the real colors.\n\n\
-                                    To fix this, follow these steps:\n  \
-                                      1. Run 'pastel colorcheck' to test if your terminal\n     \
-                                         emulator does support 24-bit colors. If this is the\n     \
-                                         case, set 'PASTEL_COLOR_MODE=24bit' to force 24-bit\n     \
-                                         mode and to remove this warning. Alternatively, make\n     \
-                                         sure that COLORTERM is properly set by your terminal\n     \
-                                         emulator.\n  \
-                                      2. If your terminal emulator does not support 24-bit\n     \
-                                         colors, set 'PASTEL_COLOR_MODE=8bit' to remove this\n     \
-                                         warning or try a different terminal emulator.\n\n\
-                                    \
-                                    For more information, see https://gist.github.com/XVilka/8346728\n");
+                            match mode {
+                                Some(ansi::Mode::Ansi8Bit) => {
+                                    if global_matches.subcommand_name() != Some("paint")
+                                        && global_matches.subcommand_name() != Some("colorcheck")
```

Comment:
I like that you moved the long print command to a new function. But why split the `if` into a `match`+`if`?

Reply:
I'm very new to rust, so I'm not really sure of how best to go about improving this. It's just my inexperience.

I believe you can match against multiple values, but I'm not sure how to include a negation in a match block, which I think you would need to replicate the above check. Would you be able to let me know what I'm missing?

Reply:
Ah, you had to change it because `mode` is now an `Option<…>`. But I think you could still simply compare to `Some(ansi::Mode::Ansi8Bit)`:
```rust
                            if mode == Some(ansi::Mode::Ansi8Bit)
                                && global_matches.subcommand_name() != Some("paint")
                                && global_matches.subcommand_name() != Some("colorcheck")
```

There's not necessarily a need for a `match` if it's a simple comparison. It would be different if you want to match on `Some(_)` or `Some(mode)` and then decide based on the `mode`.